langchain4j-azure-cosmos/src/main/java/dev/langchain4j/store/embedding/azure/cosmos/mongo/vcore/AzureCosmosDbMongoVCoreEmbeddingStore.java

```diff
@@ -0,0 +1,622 @@
+        return StreamSupport.stream(database.listCollectionNames().spliterator(), false)
+                .anyMatch(collectionName::equals);
+    }
+
+    private void createCollection(MongoDatabase database, String collectionName, CreateCollectionOptions createCollectionOptions) {
+        database.createCollection(collectionName, createCollectionOptions);
+    }
+
+    private boolean isIndexExist(String indexName) {
+        return StreamSupport.stream(collection.listIndexes().spliterator(), false)
+                .anyMatch(index -> indexName.equals(index.getString("name")));
+    }
+
+    private void createIndex(String indexName, String collectionName, MongoDatabase database) {
+        Bson commandDocument = new Document();
+        switch (this.kind) {
+            case VECTOR_IVF:
+                commandDocument = getIndexDefinitionVectorIVF(indexName, collectionName);
+                break;
+            case VECTOR_HNSW:
+                commandDocument = getIndexDefinitionVectorHNSW(indexName, collectionName);
+                break;
+        }
+
+        // Execute the BSON command
+        database.runCommand(commandDocument);
+    }
+
+    private BsonDocument getIndexDefinitionVectorIVF(String indexName, String collectionName) {
+        Document indexDefinition = new Document()
+                .append("name", indexName)
+                .append("key", new Document("embedding", "cosmosSearch"))
+                .append("cosmosSearchOptions", new Document()
+                        .append("kind", this.kind.getValue())
+                        .append("numLists", this.numLists)
+                        .append("similarity", this.similarity)
+                        .append("dimensions", this.dimensions));
+
+        // Convert the index definition to a BsonDocument
+        BsonDocument bsonIndexDefinition = indexDefinition.toBsonDocument();
+
+        // Create a BsonArray containing the index definition
+        BsonArray bsonArray = new BsonArray();
+        bsonArray.add(bsonIndexDefinition);
+
+        // Create the final BsonDocument
+        return new Document()
+                .append("createIndexes", collectionName)
+                .append("indexes", bsonArray).toBsonDocument();
+    }
+
+    private BsonDocument getIndexDefinitionVectorHNSW(String indexName, String collectionName) {
+        Document indexDefinition = new Document()
+                .append("name", indexName)
+                .append("key", new Document("embedding", "cosmosSearch"))
+                .append("cosmosSearchOptions", new Document()
+                        .append("kind", this.kind.getValue())
+                        .append("m", this.m)
+                        .append("efConstruction", this.efConstruction)
+                        .append("similarity", this.similarity)
+                        .append("dimensions", this.dimensions));
+
+        // Convert the index definition to a BsonDocument
```

Comment:
Was this code written by ChatGPT? Please remove unnecessary comments, they just polute the code

Reply:
No, it was not written using Chat GPT, but I have removed the unnecessary comments from the code.